### PR TITLE
LedgerInfo commit information aggregated in a BlockInfo struct

### PR DESCRIPTION
**Pull request by dmitri-perelman**
Summary:
The fields of LedgerInfo that describe the committed status of the Ledger are in fact
identical to the fields of the block metadata that Consensus is carrying around (the only
field of BlockInfo that is currently not present in LedgerInfo is a round).
Any update to the LedgerInfo would have to be mirrored in BlockInfo because TCB needs to verify & sign it.
Hence, this change is aggregating the LedgerInfo fields in the BlockInfo.
We had to move BlockInfo from consensus types to libra types as a result of that.

Testing: this is supposed to be a noop, existing unit test coverage

Ref #1604

### CODE REVIEW

**Comment by zekun000** [comment]
just to make sure is everyone happy with the direction?

**Comment by dmitri-perelman** [comment]
Yup. I think this one is not contentious. The PR for replacing the hash with the vote data is much less obvious :)

**Comment by aching** [comment]
Agree with @zekun000 to see how the direction of https://github.com/libra/libra/issues/1604 goes before moving forward.

File: consensus/consensus-types/src/block.rs [hunk 1]
```diff
@@ -142,17 +144,9 @@ where
         // Genesis carries a placeholder quorum certificate to its parent id with LedgerInfo
         // carrying information about version from the last LedgerInfo of previous epoch.
         let genesis_quorum_cert = QuorumCert::new(
-            VoteData::new(ancestor.clone(), ancestor),
+            VoteData::new(ancestor.clone(), ancestor.clone()),
```
Comment (zekun000) [inline consensus/consensus-types/src/block.rs]: this probably could be `BlockInfo::empty()`?

File: consensus/src/chained_bft/block_storage/pending_votes_test.rs [hunk 2]
```diff
@@ -3,20 +3,16 @@
...
+use consensus_types::{common::Round, vote::Vote, vote_data::VoteData};
 use libra_crypto::HashValue;
+use libra_types::block_info::BlockInfo;
 use libra_types::crypto_proxies::random_validator_verifier;
 use libra_types::ledger_info::LedgerInfo;
 
 fn random_ledger_info() -> LedgerInfo {
     LedgerInfo::new(
-        0,
+        BlockInfo::new(1, 0, HashValue::random(), HashValue::random(), 0, 0, None),
```
Comment (zekun000) [inline consensus/src/chained_bft/block_storage/pending_votes_test.rs]: why the epoch changes from 0 to 1? any test failure?

File: types/src/block_info.rs [hunk 3]
```diff
@@ -1,23 +1,28 @@
...
 /// This structure contains all the information needed for tracking a block
 /// without having access to the block or its execution output state. It
 /// assumes that the block is the last block executed within the ledger.
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
 pub struct BlockInfo {
     /// Epoch number corresponds to the set of validators that are active for this block.
     epoch: u64,
-    /// The consensus protocol executes proposals (blocks) in rounds, which monotically increase per epoch.
+    /// The consensus protocol is executed in rounds, which monotonically increase per epoch.
```
Comment (aching) [inline types/src/block_info.rs]: This is okay for now to make the code simpler but in the future, if we go independent signatures on LedgerInfo and QC, we'll want to probably use height for LedgerInfo and also remove any other non-essential fields for LedgerInfo.

**Pull request closed without merging**